org.eclipse.gef/src/org/eclipse/gef/internal/ui/rulers/RulerLayout.java

```diff
@@ -80,8 +85,12 @@ public void layout(IFigure container) {
 	@Override
 	public void setConstraint(IFigure figure, Object newConstraint) {
 		if (newConstraint != null && !(newConstraint instanceof Integer)) {
-			throw new IllegalArgumentException("RulerLayout was given " + newConstraint.getClass().getName() //$NON-NLS-1$
-					+ " as constraint for Figure. Integer expected!"); //$NON-NLS-1$
+			if (!(newConstraint instanceof Integer)) {
```

Comment:
I think the if is not needed anymore. we have all the checks in line 87

Reply:
That was the copy-paste error I mentioned 😉

Reply:
In any case, that should also be fixed now.